eZ/Bundle/EzPublishCoreBundle/Command/CopySubtreeCommand.php

```diff
@@ -0,0 +1,153 @@
+            ->addArgument(
+                'source-location-id',
+                InputArgument::REQUIRED,
+                'Id of subtree root location'
+            )
+            ->addArgument(
+                'target-location-id',
+                InputArgument::REQUIRED,
+                'Id of target location'
+            )
+            ->addOption(
+                'user',
+                'u',
+                InputOption::VALUE_OPTIONAL,
+                'eZ Platform username (with Role containing at least Content policies: read, versionread, edit, remove, versionremove)',
+                'admin'
+            )
+            ->setDescription('Copy subtree from one location to another');
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        parent::initialize($input, $output);
+        $this->permissionResolver->setCurrentUserReference(
+            $this->userService->loadUserByLogin($input->getOption('user'))
+        );
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @return int|null|void
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $sourceLocationId = $input->getArgument('source-location-id');
+        $targetLocationId = $input->getArgument('target-location-id');
+
+        $sourceLocation = $this->locationService->loadLocation($sourceLocationId);
+        $targetLocation = $this->locationService->loadLocation($targetLocationId);
+
+        $sourceContentType = $this->contentTypeService->loadContentType(
+            $sourceLocation->getContentInfo()->contentTypeId
+        );
+        if (!$sourceContentType->isContainer) {
+            throw new InvalidArgumentException(
+                'source-location-id',
+                'Cannot copy subtree of non container location'
+            );
+        }
+
+        $targetContentType = $this->contentTypeService->loadContentType(
+            $targetLocation->getContentInfo()->contentTypeId
+        );
+
+        if (!$targetContentType->isContainer) {
+            throw new InvalidArgumentException(
+                'target-location-id',
+                'Cannot copy location to a parent that is not a container'
+            );
+        }
+        $output->writeln(
```

Comment:
IMHO instead of displaying an warning we should ask for confirmation before execution of copying. I mean something like `Are you sure you want to copy %source% subtree (%size%) into %target%? This make take a while for big number of nested children`